RRT_star.py

```diff
@@ -1,25 +1,31 @@
+
+
+from turtle import color
 import numpy as np
 import matplotlib.pyplot as plt
 from matplotlib import patches
-from pgraph import UGraph, DGraph, UVertex
+from pgraph import UGraph, DGraph, UVertex, Edge
 import time
 from planning import RRT, Workspace, Rectangle, Circle
 import math
 
 class RRT_star(RRT):
     def __init__(self, workspace, q0):
-        super().__init__(workspace, q0)
-        self.cost_dic = {}
+        self.graph = UGraph()
+        self.workspace = workspace
+        v = self.graph.add_vertex(q0)
+        v.parent = None
+        self.v_goal = None
+        
 
     def cost(self,vb):
         """Determine the cost in between two points"""
         cost = 0
-        vb = tuple(vb)
-        for key in self.cost_dic.keys():
-            if key == vb: 
-                cost = self.cost_dic[vb]
-            # elif key == va:
-            #     cost_a = self.cost_dic[vb]
+        parent = vb.parent
+        while vb.parent != None:
+            cost += self.connection(vb.coord,parent.coord)
```

Comment:
Can you change the name of the `connection` method to `connection_cost` to make it really clear what it does?

Reply:
Sounds good!